docs/apis/streaming/event_timestamps_watermarks.md

```diff
@@ -308,6 +269,89 @@ class TimeLagWatermarkGenerator extends AssignerWithPeriodicWatermarks[MyEvent]
 </div>
 
 
+#### **With Ascending timestamps**
+
+The simplest special case for periodic watermark generation is the case where timestamps within one source occur in ascending order.
+In that case, the current timestamp can always act as a watermark, because no lower timestamps will occur any more.
+
+Note that it is only necessary that timestamps are ascending *per parallel data source instance*. For example, if
```

Comment:
Does this also holds true if a KafkaConsumer task has multiple partitions assigned?